java-frontend/src/main/java/org/sonar/java/se/constraint/ObjectConstraint.java

```diff
@@ -26,7 +26,9 @@
 public class ObjectConstraint<S extends ObjectConstraint.Status> implements Constraint {
 
   public interface Status {
-
+    default String valueAsString() {
```

Comment:
sincerely wondering about protecting/document this with unit test... You have indirect tests but what about direct one ?